tests/utils/local-network-utils/local_network_setup.go

```diff
@@ -0,0 +1,442 @@
+package localnetworkutils
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+	"os"
+	"time"
+
+	runner_sdk "github.com/ava-labs/avalanche-network-runner/client"
+	"github.com/ava-labs/avalanche-network-runner/rpcpb"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/utils/logging"
+	"github.com/ava-labs/avalanchego/utils/set"
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	"github.com/ava-labs/subnet-evm/ethclient"
+	"github.com/ava-labs/subnet-evm/plugin/evm"
+	"github.com/ava-labs/subnet-evm/rpc"
+	"github.com/ava-labs/subnet-evm/tests/utils/runner"
+	examplecrosschainmessenger "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/ExampleMessenger/ExampleCrossChainMessenger"
+	exampleerc20 "github.com/ava-labs/teleporter/abi-bindings/go/Mocks/ExampleERC20"
+	teleporterregistry "github.com/ava-labs/teleporter/abi-bindings/go/Teleporter/upgrades/TeleporterRegistry"
+	"github.com/ava-labs/teleporter/tests/utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+const (
+	fundedKeyStr = "56289e99c94b6912bfc12adc093c9b51124f0dc54ac7a766b2bc5ccf558d8027"
+)
+
+var (
+	teleporterContractAddress common.Address
+	subnetA, subnetB          ids.ID
+	subnetsInfo               map[ids.ID]*utils.SubnetTestInfo = make(map[ids.ID]*utils.SubnetTestInfo)
+	subnetNodeNames           map[ids.ID][]string              = make(map[ids.ID][]string)
+
+	fundedAddress = common.HexToAddress("0x8db97C7cEcE249c2b98bDC0226Cc4C2A57BF52FC")
+	fundedKey     *ecdsa.PrivateKey
+
+	// Internal vars only used to set up the local network
+	anrClient           runner_sdk.Client
+	manager             *runner.NetworkManager
+	warpChainConfigPath string
+)
+
+//
+// Global test state getters. Should be called within a test spec, after SetupNetwork has been called
+//
+
+func GetSubnetsInfo() []utils.SubnetTestInfo {
+	return []utils.SubnetTestInfo{
+		*subnetsInfo[subnetA],
+		*subnetsInfo[subnetB],
+	}
+}
+
+func GetTeleporterContractAddress() common.Address {
+	return teleporterContractAddress
+}
+func GetFundedAccountInfo() (common.Address, *ecdsa.PrivateKey) {
```

Comment:
I think it's strange to have the vars `fundedAddress` and `fundedKey` in this file, as the other functions have it passed in. imo the fudned address and key should live in `local_network`

Reply:
I'm going to keep them here, since that address is specific to the local network set up by the Ginkgo suite. A separate standalone local network wouldn't have the same set of funded keys, for instance.

Reply:
I'm going to keep them here, since that address is specific to the local network set up by the Ginkgo suite. A separate standalone local network wouldn't have the same set of funded keys, for instance.

Reply:
I'm going to opt to keep `local_network.go` as lightweight as possible for the time being. We can revisit later if `local_network_utils.go` proves to be too verbose.